Fix compiler warnings on Windows

A few warnings have cropped up on Windows:
```
src\opentime\rationalTime.cpp(250,34): warning C4101: 'e': unreferenced local variable
```
```
src\opentimelineio\serialization.cpp(494,50): warning C4244: 'initializing': conversion from 'const _Ty2' to 'int', possible loss of data
```

--- a/src/opentime/rationalTime.cpp
+++ b/src/opentime/rationalTime.cpp
@@ -247,7 +247,7 @@ RationalTime::from_timecode(
         seconds = std::stoi(fields[2]);
         frames  = std::stoi(fields[3]);
     }
-    catch (std::exception const& e)
+    catch (std::exception const&)
     {
         if (error_status)
         {

--- a/src/opentimelineio/serialization.cpp
+++ b/src/opentimelineio/serialization.cpp
@@ -491,7 +491,7 @@ private:
             return;
         }
 
-        const int target_version = (dg_version_it->second);
+        const int target_version = static_cast<int>(dg_version_it->second);
 
         const auto& type_rec = (
                 TypeRegistry::instance()._find_type_record(schema_name)
@@ -1005,7 +1005,7 @@ SerializableObject::Writer::write(
         // this schema
         if (target_version_it != _downgrade_version_manifest->end())
         {
-            const int target_version = target_version_it->second;
+            const int target_version = static_cast<int>(target_version_it->second);
 
             // and the current_version is greater than the target version
             if (schema_version > target_version) 

